### PR TITLE
Preventing multiple deployments from running concurrently

### PR DESCRIPTION
**Pull request by abhgupta**
@ironcladlou PTAL - my initial commit based on our discussion. I still need to do manual testing on this.

### CODE REVIEW

File: pkg/deploy/controller/deploymentconfig/factory.go [hunk 1]
```diff
@@ -48,6 +48,9 @@ func (factory *DeploymentConfigControllerFactory) Create() controller.RunnableCo
 
 	configController := &DeploymentConfigController{
 		deploymentClient: &deploymentClientImpl{
+			listDeploymentsFunc: func(namespace string) (*kapi.ReplicationControllerList, error) {
+				return factory.KubeClient.ReplicationControllers(namespace).List(labels.Everything())
```
Comment (smarterclayton) [inline pkg/deploy/controller/deploymentconfig/factory.go]: Shouldn't this be based on a label query?

**Comment by ironcladlou** [comment]
This needs reconciled with my work for replica count preservation:

https://github.com/openshift/origin/compare/master...ironcladlou:recreate-preserve-scale-factor-2?expand=1

My list implementation here already addresses @smarterclayton's comment.

**Comment by abhgupta** [comment]
Will reopen after the rebase

**Comment by abhgupta** [comment]
Rebased onto @ironcladlou 's changes

**Comment by abhgupta** [comment]
Created https://github.com/openshift/origin/pull/2109

**Pull request closed without merging**